src/main/java/org/springframework/data/elasticsearch/core/SearchOperations.java

```diff
@@ -170,32 +202,203 @@ default long count(Query query, IndexCoordinates index) {
+	 */
+	List<AggregatedPage<? extends SearchHit<?>>> searchForPage(List<? extends Query> queries, List<Class<?>> classes,
+			IndexCoordinates index);
+
+	/**
+	 * Execute the criteria query against elasticsearch and return result as {@link List}
+	 *
+	 * @param <T> element return type
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return list of found objects
+	 */
+	<T> SearchHits<T> searchForList(Query query, Class<T> clazz, IndexCoordinates index);
+
+	/**
+	 * Execute the multi search query against elasticsearch and return result as {@link List}
+	 *
+	 * @param queries the queries to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @param <T> element return type
+	 * @return list of found objects
+	 */
+	default <T> List<List<SearchHit<T>>> searchForList(List<Query> queries, Class<T> clazz, IndexCoordinates index) {
+		return searchForPage(queries, clazz, index).stream().map(Page::getContent).collect(Collectors.toList());
+	}
+
+	/**
+	 * Execute the multi search query against elasticsearch and return result as {@link List}
+	 *
+	 * @param queries the queries to execute
+	 * @param classes the entity classes used for property mapping
+	 * @param index the index to run the query against
+	 * @return list of list of found objects
+	 */
+	default List<List<?>> searchForList(List<Query> queries, List<Class<?>> classes, IndexCoordinates index) {
+		return searchForPage(queries, classes, index).stream().map(Page::getContent).collect(Collectors.toList());
+	}
+
+	/**
+	 * Execute the query against elasticsearch and return ids
+	 *
 	 * @param query the query to execute
 	 * @param clazz the entity clazz used for property mapping
 	 * @param index the index to run the query against
+	 * @return list of found object ids
+	 */
+	List<String> searchForIds(Query query, Class<?> clazz, IndexCoordinates index);
+
+	/**
+	 * more like this query to search for documents that are "like" a specific document.
+	 *
 	 * @param <T> element return type
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
 	 * @return page with the results
 	 */
-	<T> Page<T> moreLikeThis(MoreLikeThisQuery query, Class<T> clazz, IndexCoordinates index);
+	<T> AggregatedPage<SearchHit<T>> searchMoreLikeThis(MoreLikeThisQuery query, Class<T> clazz, IndexCoordinates index);
+
+	/**
+	 * Returns scrolled page for given query
+	 *
+	 * @param scrollTimeInMillis duration of the scroll time
+	 * @param query The search query.
+	 * @param clazz The class of entity to retrieve.
+	 * @param index the index to run the query against
+	 * @return scrolled page result
+	 */
+	<T> ScrolledPage<SearchHit<T>> searchScrollStart(long scrollTimeInMillis, Query query, Class<T> clazz,
```

Comment:
How about not introducing these methods yet as we should have `SearchSlice` in place first. See also above my comment that scroll methods have a rather internal flavor and we should be able to just change `startScroll`/`continueScroll` return types.

Reply:
We need the `SearchHit` variant of these for the `public <T> CloseableIterator<SearchHit<T>> searchForStream(Query query, Class<T> clazz, IndexCoordinates index)`method to be able to return a stream of `SearchHit` objects. If we change the return of `startScroll/continueScroll` later, we cannot change the stream now.

Reply:
Okay. Let's revisit the `search…Scroll` methods at a later stage, once we have all result and slice types in place.